django/db/models/query.py

```diff
@@ -1534,8 +1534,11 @@ def __init__(self, lookup, queryset=None, to_attr=None):
         self.prefetch_through = lookup
         # `prefetch_to` is the path to the attribute that stores the result.
         self.prefetch_to = lookup
-        if queryset is not None and not issubclass(queryset._iterable_class, ModelIterable):
-            raise ValueError('Prefetch querysets cannot use values().')
+        if queryset is not None:
+            if not hasattr(queryset, '_iterable_class'):
+                raise AttributeError("Prefetch queryset should has an '_iterable_class' attribute.")
```

Comment:
We shouldn't raise errors about private APIs, we can error nicely when using `Prefetch` with `raw()`, e.g.
```python
if queryset is not None:
    if isinstance(queryset, RawQuerySet) or (
        hasattr(queryset, '_iterable_class') and not issubclass(queryset._iterable_class, ModelIterable)
    ):
        raise AttributeError('Prefetch querysets cannot use raw() and values().')
```

Reply:
replaced you suggested patch and changed `AttributeError` with `ValueError`.

Reply:
Yes, `ValueError` sorry.